src/ansys/speos/core/sensor.py

```diff
@@ -3742,3 +3742,617 @@ def set_geometries(self, geometries: [List[GeoRef]]) -> Sensor3DIrradiance:
+        elif template.HasField("intensity_orientation_x_as_meridian"):
+            return template.intensity_orientation_x_as_meridian.intensity_dimensions.y_start
+
+    @y_start.setter
+    def y_start(self, value: float):
+        template = self._sensor_template.intensity_sensor_template
+        if template.HasField("intensity_orientation_conoscopic"):
+            raise TypeError("Conoscopic Sensor has no y_start dimension")
+        elif template.HasField("intensity_orientation_x_as_parallel"):
+            template.intensity_orientation_x_as_parallel.intensity_dimensions.y_start = value
+        elif template.HasField("intensity_orientation_x_as_meridian"):
+            template.intensity_orientation_x_as_meridian.intensity_dimensions.y_start = value
+
+    @property
+    def y_sampling(self) -> int:
+        """Sampling along y-axis.
+
+        By default, ``100``.
+        """
+        template = self._sensor_template.intensity_sensor_template
+        if template.HasField("intensity_orientation_conoscopic"):
+            return None
+        elif template.HasField("intensity_orientation_x_as_parallel"):
+            return template.intensity_orientation_x_as_parallel.intensity_dimensions.y_sampling
+        elif template.HasField("intensity_orientation_x_as_meridian"):
+            return template.intensity_orientation_x_as_meridian.intensity_dimensions.y_sampling
+
+    @y_sampling.setter
+    def y_sampling(self, value: int):
+        template = self._sensor_template.intensity_sensor_template
+        if template.HasField("intensity_orientation_conoscopic"):
+            raise TypeError("Conoscopic Sensor has no y_sampling dimension")
+        elif template.HasField("intensity_orientation_x_as_parallel"):
+            template.intensity_orientation_x_as_parallel.intensity_dimensions.y_sampling = value
+        elif template.HasField("intensity_orientation_x_as_meridian"):
+            template.intensity_orientation_x_as_meridian.intensity_dimensions.y_sampling = value
+
+    @property
+    def theta_max(self) -> float:
+        """Maximum theta angle on consocopic type (in deg).
+
+        By default, ``45``.
+        """
+        template = self._sensor_template.intensity_sensor_template
+        if template.HasField("intensity_orientation_conoscopic"):
+            return (
+                template.intensity_orientation_conoscopic.conoscopic_intensity_dimensions.theta_max
+            )
+        else:
+            return None
+
+    @theta_max.setter
+    def theta_max(self, value: float):
+        template = self._sensor_template.intensity_sensor_template
+        if template.HasField("intensity_orientation_conoscopic"):
+            template.intensity_orientation_conoscopic.conoscopic_intensity_dimensions.theta_max = (
+                value
+            )
+        else:
+            raise TypeError("Only Conoscopic Sensor has theta_max dimension")
+
+    @property
+    def theta_max_sampling(self) -> int:
```

Comment:
max_sampling maybe a bit confusing? theta_sampling?

Reply:
good catch will refactor it